pkg/middleware/rate_limit_test.go

```diff
@@ -0,0 +1,87 @@
+package middleware
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/grafana/grafana/pkg/models"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/macaron.v1"
+)
+
+type execFunc func() *httptest.ResponseRecorder
+type advanceTimeFunc func(deltaTime time.Duration)
+type rateLimiterScenarioFunc func(c execFunc, t advanceTimeFunc)
+
+func rateLimiterScenario(t *testing.T, desc string, rps int, burst int, fn rateLimiterScenarioFunc) {
+	t.Run(desc, func(t *testing.T) {
+		m := macaron.New()
+		defaultHandler := func(c *models.ReqContext) {
+			resp := make(map[string]interface{})
+			resp["message"] = "OK"
+			c.JSON(200, resp)
+		}
```

Comment:
Nit, but would you mind moving `defaultHandler` to line 36 (i.e. before the `m.Get` call)? It makes it more obvious to a reader (such as myself) that `defaultHandler` isn't involved in the initialization of `m`. It also has the benefit of grouping `m` initialization calls.

Reply:
Done! Thanks a lot for the review @aknuds1  :)